src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -88,30 +101,75 @@
 from snowflake.connector import DictCursor, ProgrammingError
 from snowflake.connector.cursor import SnowflakeCursor
 
+ApplicationPackageChildrenTypes = (
+    StreamlitEntityModel | FunctionEntityModel | ProcedureEntityModel
+)
+
+
+class ApplicationPackageChildIdentifier(UpdatableModel):
+    schema_: Optional[str] = Field(
+        title="Child entity schema", alias="schema", default=None
+    )
+
+
+class EnsureUsableByField(UpdatableModel):
+    application_roles: Optional[Union[str, Set[str]]] = Field(
+        title="One or more application roles",
+        default=None,
+    )
+
+    @field_validator("application_roles", mode="before")
+    @classmethod
+    def ensure_app_roles_is_a_set(
+        cls, application_roles: Optional[Union[str, Set[str]]]
+    ) -> Optional[Union[str, Set[str]]]:
+        if isinstance(application_roles, str):
+            return set(application_roles)
+        return application_roles
+
+
+class ApplicationPackageChildField(UpdatableModel):
+    target: str = Field(title="The key of the entity to include in this package")
+    ensure_usable_by: Optional[EnsureUsableByField] = Field(
+        title="Use to automatically grant USAGE privilege on the child object",
+        default=None,
+    )
+    application_roles: Optional[Union[str, Set[str]]] = Field(
+        title="The application role to be granted usage on this child entity",
+        default=None,
+    )
+    identifier: ApplicationPackageChildIdentifier = Field(
```

Comment:
are we supporting overrites on name/db as well? (I thought we're only supporting schema updates in the beginning)

Reply:
We're only supporting schema overrides for now: `ApplicationPackageChildIdentifier` only has `schema` field.
We will allow to override `name` in the future, but not `database` as we are always using the application as the database.